compiler/vmgen.nim

```diff
@@ -1767,13 +1767,6 @@ proc gen(c: PCtx; n: PNode; dest: var TDest; flags: TGenFlags = {}) =
     of skConst:
       let constVal = if s.ast != nil: s.ast else: s.typ.n
       gen(c, constVal, dest)
-    of skEnumField:
```

Comment:
mainly because it looks like we never get here - afaict `skEnumField` gets converted to an int node in `semfold.nim` leading to this branch never being taken.. can add it back, though lit looks pretty dead.

Reply:
It's more robust to not assume the frontend did this transformation to the underlying integer value. Don't remove this code.

Reply:
sure,  added a comment instead. 

fyi, the js backend doesn't handle it either and there's a number of other `sk...` cases that get the `internalError` treatment here through an else branch.